core/runtime/src/main/java/io/quarkus/runtime/annotations/RegisterForProxy.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.runtime.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation that can be used to force an interface (including its super interfaces) to be registered for dynamic proxy
+ * generation in native image mode.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@Repeatable(RegisterForProxy.List.class)
+public @interface RegisterForProxy {
+
+    /**
+     * Alternative interfaces that should actually be registered for dynamic proxy generation instead of the current interface.
+     * This allows for interfaces in 3rd party libraries to be registered without modification or writing an
+     * extension. If this is set then the interface it is placed on is not registered for dynamic proxy generation, so this
+     * should generally just be placed on an empty interface that is not otherwise used.
+     */
+    Class<?>[] targets() default {};
```

Comment:
@gastaldi I chose `targets` because `@RegisterForReflection` has similar method `targets` which is basically achieving the same thing:

https://github.com/quarkusio/quarkus/blob/73d9baef74aaddc8aaadcec16e69bf71a92dce2e/core/runtime/src/main/java/io/quarkus/runtime/annotations/RegisterForReflection.java#L37-L44